### PR TITLE
Forward objectsDidChange to dependency sync

### PR DESCRIPTION
**Pull request by basine**
We were not forwarding changed objects to the dependency sync. Which meant that if an object had a dependency when it was added, the dependency was never removed when the dependent object changed.

e.g. A message fails to send due to missing clients and is added to the objects with dependencies. When the missing clients are fetched, the dependency is resolved. However, if objectsDidChange is never called, the message will never be resent.

### CODE REVIEW

File: Source/Transcoders/GenericMessageRequestStrategy.swift [hunk 1]
```diff
@@ -49,7 +49,7 @@ extension GenericMessageEntity : EncryptedPayloadGenerator {
     
 }
 
-public class GenericMessageRequestStrategy : OTREntityTranscoder<GenericMessageEntity>, ZMRequestGenerator {
+public class GenericMessageRequestStrategy : OTREntityTranscoder<GenericMessageEntity>, ZMRequestGenerator, ZMContextChangeTracker {
```
Comment (typfel) [inline Source/Transcoders/GenericMessageRequestStrategy.swift]: Could be a `ZMContextChangeTrackerSource` which returns the `sync` instance

Comment (basine) [inline Source/Transcoders/GenericMessageRequestStrategy.swift]: not sure if that really makes a difference. In both cases you would have to somehow register them in the parent strategy (I am working on a PR for zmessaging, see https://github.com/wireapp/wire-ios-sync-engine/pull/265/files

Comment (basine) [inline Source/Transcoders/GenericMessageRequestStrategy.swift]: Actually making it a source makes it harder in that case.

Comment (typfel) [inline Source/Transcoders/GenericMessageRequestStrategy.swift]: Ah right forgot it was used inside another strategy

**Pull request merged**